### ISSUE TITLE
list/grid view of tags in "Tag Details" pane

### ISSUE THREAD

**Issue by phraenquex** [m2ms/fragalysis-frontend#1102]
Add a a "list / grid" toggle in Tag details?  To rearrange the icons like we have in file explorers.  
* "list" will be the curernt view
* "grid" will arrange tag icons in a grid, as many columns as fit in the panel width.  
(As in attached images.)

![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/96abf694-c393-4c82-ae92-a6a0a4d4a9e9) 

![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/2e97412f-310a-4d31-a551-79e6748cb410)

**Comment by RoboMatuska** [comment]
@phraenquex  I created switch button for Tag details, but I need you opinion which visual is better
v1 :
![image](https://github.com/m2ms/fragalysis-frontend/assets/132336191/63341108-2d49-432c-abaf-3807a4fa00fe)
or
v2 with smaller width
![image](https://github.com/m2ms/fragalysis-frontend/assets/132336191/0ab1d5d7-5662-4f70-bcae-5519f371523a)
 or v3 with absolute tag length
![image](https://github.com/m2ms/fragalysis-frontend/assets/132336191/bfab4306-4efb-4ea5-a0f4-eb2c66bafc4d)

Always depends on the length of the text in the tag

**Comment by RoboMatuska** [comment]
Yes of course, v1 = 150px, v2 = 90px and v3 = 210px, but always depends  on length of the text. If is short tag name its ok for one row, but if is long tag name we have 2/3 and more rows for one tag and final grid look isn't very well.

**Comment by phraenquex** [comment]
Hmm.  The ideal solution would allow me to change the width dynamically - is that easy, or a lot of work?

But for now, use option 1 (three columns), and add these fixes:
* remove the tick-mark for selected tags (the new colouring means it's no longer necessary)
* make all tag boxes the same size
* middle-align the text (currently it's top-aligned)

Then we have to check how this behaves for different window sizes.

**Comment by phraenquex** [comment]
Thanks @RoboMatuska - then elts not do the dynamic changes.

**Comment by phraenquex** [comment]
@RoboMatuska looks great!  In grid-view, reduce the width to 50% of the current width.

**If easy(!!)**:  if any string is too long, then text-wrap the string, and make _all_ tags double-height, and middle-align the text.
_If not easy_ - the current solution works well (using ... and the tooltip having the full info).

**Comment by phraenquex** [comment]
Also, when I toggle, don't change the panel height - just leave it as is.  

(Making it taller is okay-ish, i.e. grid->list.  But making it smaller (list->grid) is rather distracting.  So better not to touch the height.)

**Comment by phraenquex** [comment]
@RoboMatuska on second thoughts, don't change the resizing:  after I played with it in staging, I see now that the resizing is really helpful.

**Comment by phraenquex** [comment]
@RoboMatuska on second thoughts, don't change the resizing of the panel height.  after I played with it in staging, I see now that the resizing is really helpful.  (I hid the comment that I made, suggesting getting rid of the resizing.)

But DO look at the tag width comment, in case that's easy.  

Also, the vertical alignment is still not perfect (see [this snapshot](https://fragalysis.xchem.diamond.ac.uk/viewer/react/projects/3259/2778)): 
       ![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/beffae51-6774-4a09-8838-34129ccd70c9)

What's meant to happen is this:
![image](https://github.com/m2ms/fragalysis-frontend/assets/33267937/ac0e930c-0892-4304-8d82-0a351d1a33e7)

**Comment by phraenquex** [comment]
@boriskovar-m2ms snapshots don't preserve grid/list layout.  (Not a blocker, so move to a new ticket if it's too difficult to solve now.)

**Comment by phraenquex** [comment]
These are definitely less important than fixing #741 .
